### PR TITLE
feat(sbb-link): implement LinkButtonProperties interface

### PR DESCRIPTION
**Pull request by DavideMininni-Fincons**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures

## Changes

Changes in this pull request:

- src/global/interfaces/link-button-properties.ts
    - two new interface have been added (`LinkProperties` and `ButtonProperties`); each of them has some required attribute (eg. `href` for link and `name` for button); 
    - the `LinkButtonProperties` interface extends both, plus the `AccessibilityProperties` interface;
    - methods `getLinkButtonBaseAttributeList(...)` and `getLinkButtonAttributeList(...)` have now a safe check to avoid undefined attributes (eg. class='undefined'), see also [this link](https://github.com/tc39/proposal-object-rest-spread/issues/45);
- src/components/sbb-link/sbb-link.tsx
    - the component now implements `LinkButtonProperties`;
    - `icon` prop has been renamed to `iconName` to be more consistent;
    - props have been reordered: 
        - required props due first point of previous list have also 'undefined' as type (eg. `@Prop() public href: string | undefined;`);
        - optional props have no default anymore (eg. since `iconPlacement` is used only if an icon is provided and since there's no CSS rule for `start` value, the default `start` value has been removed);
- src/components/sbb-link/sbb-link.stories.js: parameters have been reordered and categorized;
- src/components/sbb-link/sbb-link.spec.ts: test fixed with icon cases;
- src/components/sbb-link/sbb-link.custom.d.ts: no need for optional modifier.

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [x] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

`icon` property in sbb-link is now `iconName`

### CODE REVIEW

File: src/components/sbb-link/sbb-link.tsx [hunk 1]
```diff
@@ -25,195 +25,154 @@ import { hostContext } from '../../global/helpers/host-context';
   styleUrl: 'sbb-link.scss',
   tag: 'sbb-link',
 })
-export class SbbLink implements AccessibilityProperties, ComponentInterface {
+export class SbbLink implements LinkButtonProperties, ComponentInterface {
+  @Element() public el!: HTMLElement;
```
[the remark this answers is not in the document]
Comment (DavideMininni-Fincons) [reply src/components/sbb-link/sbb-link.tsx]: I didn't noticed it.. fixed & pushed 💪

File: src/components/sbb-link/sbb-link.tsx [hunk 2]
```diff
@@ -25,195 +25,154 @@ import { hostContext } from '../../global/helpers/host-context';
...
+  /**
+   * Generate the class attribute based on component's parameters.
+   */
   private _getClassString(): string {
     const textSizeClass = this.variant === 'inline' ? '' : ` sbb-link--text-${this.textSize}`;
     const iconPositionClass =
-      this.iconPlacement === 'start'
-        ? ' sbb-link--icon-placement-start'
-        : ' sbb-link--icon-placement-end';
+      this.iconPlacement === 'end' ? ` sbb-link--icon-placement-${this.iconPlacement}` : '';
```
Comment (dauriamarco) [inline src/components/sbb-link/sbb-link.tsx:151]: At this point we could further refactor the property by making it as a boolean: "placeIconOnEnd" (?)

**Review by dauriamarco — CHANGES_REQUESTED** [review]

File: src/components/sbb-link/sbb-link.tsx (hunk 2 above)
Comment (DavideMininni-Fincons) [reply src/components/sbb-link/sbb-link.tsx]: Done in commit 9f3ac8a572b2aa5cc8642f308da86c315302837d

**Review by dauriamarco — APPROVED** [review]
LGTM 👍

**Pull request merged**